src/pyfaf/storage/events.py

```diff
@@ -36,3 +37,16 @@ def before_delete(mapper, connection, target):
     for lobname in target.__lobs__:
         if target.has_lob(lobname):
             target.del_lob(lobname)
+
+
+@event.listens_for(Session, "before_flush")
+def store_semantic_version_for_build(session, flush_context, instances):
+    """
+    Store semnatic version (Build.semver) converted from text version
+    (Build.version)
+    """
+
+    objects = session.new.union(session.dirty)
+    for obj in [c for c in objects if isinstance(c, Build)]:
+        if isinstance(obj, Build):
```

Comment:
I guess the property setter would mess with the ORM too much.
What about http://docs.sqlalchemy.org/en/rel_0_7/orm/events.html#sqlalchemy.orm.events.AttributeEvents.set ? This event seems a little more fine-grained so it might have less performance impact.